src/strings.cpp

```diff
@@ -363,6 +363,18 @@ void SetDParamStr(size_t n, const std::string &str)
 	_global_string_params.SetParam(n, str);
 }
 
+/**
+ * This function is used to "bind" the std::string to a OpenTTD dparam slot.
+ * Contrary to the other \c SetDParamStr functions, this moves the string into
+ * the parameter slot.
+ * @param n slot of the string
+ * @param str string to bind
+ */
+void SetDParamStr(size_t n, std::string &&str)
+{
+	_global_string_params.SetParam(n, std::move(str));
```

Comment:
Is it valid to use std::move here? str might not be a temporary.

Reply:
https://godbolt.org/z/nhr8d74ch

If there is only a `std::string &&`, then passing non-rvalues will yield a compilation error... unless you explicitly use `std::move`.
If there is only a `const std::string &`, then thing will just be as the were before.
With this change there is a `const std::string &` and `std::string &&`, which means that for rvalues the latter is called which does the `std::move` to make it implicit that it should be passed on. The former remains as is.

So, the only way to pass a `std::string` that should not be copied to `SetDParamStr` is when it's either a (temporary) rvalue, or when you explicitly `std::move` it. It's practically the same construct that is used for copy vs move constructors/assignments.